server/postgres/src/storage.ts

```diff
@@ -1148,6 +1157,14 @@ abstract class PostgresAdapterBase implements DbAdapter {
             }
             break
           case '$nin':
+            if (Array.isArray(val) && val.includes(null)) {
+              res.push(
+                `(${tkey} NOT IN (${val
+                  .filter((it) => it != null)
+                  .map((v: any) => `'${v}'`)
+                  .join(', ')}) AND ${tkey} IS NOT NULL)`
+              )
+            }
             if (val.length > 0) {
```

Comment:
should it be in `else` block?

Reply:
Yep, it shoudl be in else block. Thanks.